### PR TITLE
Restructure the folders by technology

### PR DESCRIPTION
**Pull request by cansavvy**
## Purpose:
Related to #118 and #102 and #111 

In this PR, I'm getting set to restructure everything by what the respective 'chapters' will be for the tutorial. 

Here's what's happened in this PR: 
- I got rid of all the GenePattern sections. 
- I got rid of the `compare-processing` module. 
- I put everything in folders according to what "section" they should be in 
- Every section has its own intro document Rmd that will eventually be filled in with context info. 
- I separate RNA-seq and microarray `differential-expression` modules and their associated files and READMEs into their `differential-expression` modules in their respective section. 
- I went through all the READMEs and fixed the URL links to go to their new places.
- I fixed the .gitignore for the new locations.

### Questions for review: 
- How do we like this structure? 
- Since we will be adding more context and "easing into it" steps toward the analyses, I've gone ahead and figured we don't want the GenePattern sections anymore but let me know if this is not what we want. 
- Did I miss any links or files that are misplaced?

### CODE REVIEW

**Comment by cansavvy** [comment]
So GitHub doesn't seem to register the changes in the READMEs very well, but I can confirm that the READMEs in each module only change by their links and getting rid of GenePattern mentions. So just focus on those items. The content of the material of the READMEs are going to change completely and/or be deleted in the future anyway, but in the meanwhile I tried to keep them consistent with the file moves.

File: 01-getting-started/getting-started.Rmd [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (cbethell) [inline 01-getting-started/getting-started.Rmd:1]: Do you have an idea of what the contents of this notebook will look like? Also will this `01-getting-started` directory have a `README.md` file as does the other example analysis directories?

File: .gitignore [hunk 2]
```diff
@@ -1,27 +1,24 @@
...
 .Rhistory
-.DS_Store
-data/.DS_Store
-differential-expression/.DS_Store
-
-# Ignore ID refinery mappings for irrelevant platforms
-compare-processing/data/id_refinery/all_1536267482.zip
 
 # This data is downloaded from another part of the repo in essence
-normalize-own-data/data/GSE39842_RMA_converted.tsv
```
Comment (cbethell) [inline .gitignore:12]: Looks like this file also needs to be removed from this file path (forgive me if I am wrong, I checked out your branch and still see the file at this path but it is quite possible that it is just local).

File: 01-getting-started/getting-started.Rmd (hunk 1 above)
Comment (cbethell) [reply 01-getting-started/getting-started.Rmd:1]: Ah the answer to my question is in alexslemonade/refinebio-examples/issues/109 👍

File: 02-microarray/intro-to-microarray.Rmd [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+title: "Introduction to Microarray"
```
[the remark this answers is not in the document]
Comment (cansavvy) [reply 02-microarray/intro-to-microarray.Rmd:2]: Yes. It's TBD, but we'll want to link it in the website header, so I've included blank files for now.

File: .gitignore (hunk 2 above)
Comment (cansavvy) [reply .gitignore:12]: I think I fixed this?

Comment (cbethell) [reply .gitignore:12]: Looks like it 👍

**Comment by cansavvy** [comment]
We can't merge this until https://github.com/AlexsLemonade/refinebio-frontend/issues/907 is resolved.

**Comment by cansavvy** [comment]
We need to get these PRs merged before I merge this PR:
- [ ] https://github.com/AlexsLemonade/refinebio-frontend/pull/908
- [ ] https://github.com/AlexsLemonade/refinebio-docs/pull/135

**Comment by cansavvy** [comment]
User facing links have been changed to release/v1.0, so this is good to go.

**Comment by jaclyn-taroni** [comment]
Frontend hasn't been deployed yet?

**Comment by cansavvy** [comment]
> Frontend hasn't been deployed yet?

This was my bad. https://github.com/AlexsLemonade/refinebio-frontend/pull/913 should have also bee on my list above.

**Pull request merged**